src/syntax/Rewrite.jl

```diff
@@ -31,787 +31,930 @@ function block_to_sf(block)
 
-      if dyvar_name ∉ keys(R_dict)
-        new_dyvar = deepcopy(dyvar)
-        push!(R_block.dyvars, new_dyvar)
-        push!(R_dict, dyvar_name => new_dyvar)
-      else
-        new_dyvar = R_dict[dyvar_name]
+        if (original_inflow != :F_NONE)
+          link = (original_inflow, object_name)
+          add_link_if_not_already!(L_connect_dict, link, :I)
+          add_link_if_not_already!(remove_connect_dict, link, :I)
+        end
+        if (original_outflow != :F_NONE)
+          link = (original_outflow, object_name)
+          add_link_if_not_already!(L_connect_dict, link, :O)
+          add_link_if_not_already!(remove_connect_dict, link, :O)
+        end
       end
-      replace!(new_dyvar[2].args, src => tgt)
-    end
+
+      # needs to have dyvar added...
+      push!(R_flow_queue, object)
+      # end
+
+
+    end 
+
+
   end
 end
 
-"""
-Given a dyvar, recursively add all dyvars contained in it to R.
-"""
-function recursively_add_dyvars_R!(current_dyvar, sf_block, new_block, new_dict, name_dict)
-  dyvar_name = current_dyvar[1]
-  dyvar_copy = deepcopy(current_dyvar)
-  dyvar_definition = dyvar_copy[2]
-  if dyvar_name ∉ keys(new_dict)
-    push!(new_block.dyvars, dyvar_copy)
-    push!(new_dict, dyvar_name => dyvar_copy)
-  end
-
-  # For everything not yet in the dict, add to dict
-  for object ∈ filter(∉(keys(new_dict)), dyvar_definition.args[2:end])
-    object_type = name_dict[object][1]
-    object_index = name_dict[object][2]
-    object_definition = get_from_correct_vector(sf_block, object_index, object_type)
-    if object_definition isa Symbol  
-      push!(new_dict, object => (object_definition,))
-    else
-      push!(new_dict, object => object_definition)
+
+
+
+function remove_from_flows!(object_name, sf_block, L, L_set, name_dict, L_connect_dict, remove_connect_dict)
+  for flow in sf_block.flows
+    inflow_stock = flow[3]
+    outflow_stock = flow[1]
+    flow_name = flow[2].args[1]
+    flow_dyvar = flow[2].args[2]
+
+    link = (object_name, flow_name)
+
+    if (inflow_stock == object_name) || (outflow_stock == object_name)
+      add_part_if_not_already!(L_set, L, flow_dyvar, :V, name_dict[flow_dyvar].index, sf_block)
+      add_part_if_not_already!(L_set, L, flow_name, :F, name_dict[flow_name].index, sf_block)
     end
 
-    add_to_correct_vector!(new_block, object_definition, object_type)
+    if (inflow_stock == object_name)
+      add_link_if_not_already!(L_connect_dict, link, :I)
+      add_link_if_not_already!(remove_connect_dict, link, :I)
+    end
 
+    if (outflow_stock == object_name)
+      add_link_if_not_already!(L_connect_dict, link, :O)
+      add_link_if_not_already!(remove_connect_dict, link, :O)
+    end
 
-    if object_type == :V
-      recursively_add_dyvars_R!(object_definition, sf_block, new_block, new_dict, name_dict)
+
+  end
+
+end
+
+
+function remove_block!(removed, removed_set, L_set, name_dict, L, sf_block, L_connect_dict, remove_connect_dict)
```

Comment:
We could have a discussion on this.  Some of the mutation is probably unnecessary, but I don't think it can be completely avoided.  I don't know what a better way to do this would be.

We need to construct three stockflows, which I do by iterating over each line of the block and determining what needs to be added to each.

I do empathize with the whole 'hard to reason about correctness'.  It's why I overcompensated with test cases.

I still think this is an improvement over the previous version.